cmd/otel-allocator/collector/collector_test.go

```diff
@@ -0,0 +1,96 @@
+package collector
+
+import (
+	"context"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	collectors := []string{}
+
+	client := Client{
+		k8sClient:     fake.NewSimpleClientset(),
+		collectorChan: make(chan []string),
+	}
+
+	labelMap := map[string]string{
+		"app.kubernetes.io/instance":   "default.test",
+		"app.kubernetes.io/managed-by": "opentelemetry-operator",
+	}
+
+	opts := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labelMap).String(),
+	}
+
+	client.Watch(ctx, labelMap, func(collectorList []string) {})
+	// adding sleep to alow the collector watch function to start
+	time.Sleep(1 * time.Second)
```

Comment:
This is probably an indication that your code needs a callback, to notify when it's ready. Waiting N seconds will make this test either inefficient, as it waits more than it needs, or intermittent when it takes longer than 1s for the client to be ready.

Reply:
Yes I agree. I have replaced it with sync.WaitGroup which is a better way to test I feel.